### PR TITLE
Delete flaky test RestoreNetCore_MultipleProjects_SameToolDifferentVersions_NoOp_FailsAsync

### LINKED ISSUE #9100: Test: RestoreNetCore_MultipleProjects_SameToolDifferentVersions_NoOp_FailsAsync is flaky
[This assertion](https://github.com/NuGet/NuGet.Client/blob/ac177820b30d328e3c6a4f4a6801b28cc7441a66/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs#L1877) is sometimes failing with: 

> 8 needs to be 9 or 10

edit: as per @nkolev92's comment below, this test is redundant and flaky, so is being deleted.

### PR DESCRIPTION
**Pull request by zivkan**
Fixes: NuGet/Home#9100

### CODE REVIEW

**Review by nkolev92 — COMMENTED** [review]
See: https://github.com/NuGet/Home/issues/9100#issuecomment-578281497

I'd delete the test tbh.

I don't see it as super relevant as the other tests kind of ensure this is happening.

tldr;
bad test.

**Review by nkolev92 — APPROVED** [review]

**Review by rrelyea — APPROVED** [review]
I don't have the context on the test. But yes, if it is flaky, and we want to delete it, this change will do that.
